website/project/views/node.py

```diff
@@ -856,7 +856,7 @@ def private_link_table(node, **kwargs):
 @must_be_valid_project
 def get_editable_children(auth, node, **kwargs):
 
-    if not node.can_edit(auth):
+    if not node.has_permission(auth.user, 'admin'):
```

Comment:
That's right. Read only users can't make VOL, and shouldn't be able to. So
the side effect described is desired.

On Wed, Sep 23, 2015 at 2:56 PM, Steven Loria <notifications@github.com>
wrote:

> In website/project/views/node.py
> <https://github.com/CenterForOpenScience/osf.io/pull/4342#discussion_r40242810>
> :
>
> > @@ -856,7 +856,7 @@ def private_link_table(node, **kwargs):
> >  @must_be_valid_project
> >  def get_editable_children(auth, node, **kwargs):
> >
> > -    if not node.can_edit(auth):
> > +    if not node.has_permission(auth.user, 'admin'):
>
> ^ if we do that, then I think we can just add @must_have_permission(WRITE)
> (rather than ADMIN) to the view. One side-effect is that users will only be
> able to create VOLs for child nodes that the current user has admin access
> to. I think this is the desired behavior anyway. @jlcohoon
> <https://github.com/jlcohoon> correct me if I'm wrong.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CenterForOpenScience/osf.io/pull/4342/files#r40242810>
> .
>